content/docs/user-guide/experiment-management/hydra-composition.md

```diff
@@ -0,0 +1,260 @@
+└── train
+    ├── baseline.yaml
+    ├── model
+    │   ├── alexnet.yaml
+    │   ├── efficientnet.yaml
+    │   └── resnet.yaml
+    └── optimizer
+        ├── adam.yaml
+        └── sgd.yaml
+```
+
+Along with a file defining the top-level
+[Hydra Defaults List](https://hydra.cc/docs/tutorials/basic/your_first_app/defaults/)
+(`conf/config.yaml`):
+
+```dvc
+$ cat conf/config.yaml
+defaults:
+  - dataset: imagenette
+  - train/model: resnet
+  - train/optimizer: SGD
+```
+
+<admon type="info">
+
+You can set the [`dvc config hydra`](/doc/command-reference/config#hydra)
+options to provide custom locations for the _Config Group_ and the top-level
+_Defaults List_.
+
+</admon>
+
+The main difference with the official tutorial is that, instead of relying on
+the
+[`@hydra.main`](https://hydra.cc/docs/tutorials/basic/your_first_app/simple_cli/)
+Python decorator, DVC will take care of composing and dumping the configuration
+so we can instead rely on a single file: `params.yaml`.
+
+## Setting up DVC
+
+We can enable the Hydra composition by running:
+
+```dvc
+$ dvc config hydra.enabled True
+```
+
+### Testing the composition
+
+To check how the composition works, we can create a dummy `dvc.yaml` that prints
+the contents of `params.yaml`:
+
+```yaml
+stages:
+  print-params:
+    cmd: cat params.yaml
+```
+
+Now, if we run an experiment, we will see the composed configuration printed:
+
+```dvc
+$ dvc exp run
+Running stage 'print-params':
+> cat params.yaml
+dataset:
+  url: https://s3.amazonaws.com/fast-ai-imageclas/imagenette2-160.tgz
+  output_folder: imagenette
+train:
+  model:
+    name: ResNet
+    size: 50
+    weights: ResNet50_Weights.IMAGENET1K_V2
+  optimizer:
+    name: SGD
+    lr: 0.001
+    momentum: 0.9
+```
+
+### Building a Pipeline
+
+To showcase a more realistic use case, we are going to build a Pipeline with 2
+stages.
+
+The first stage downloads the dataset, and depends on the
+<abbr>parameters</abbr> defined in the `dataset` section:
+
+```yaml
+stages:
+  setup-dataset:
+    cmd:
+    - wget ${dataset.url} -O tmp.tgz
+    - mkdir -p ${dataset.output_folder}
+    - tar zxvf tmp.tgz -C ${dataset.output_folder}
+    - rm tmp.tgz
+    outs:
+      - ${dataset.output_folder}
+  ...
+```
+
+We use [Templating](/doc/user-guide/project-structure/dvcyaml-files#templating)
+in order to configure our shell commands (`mkdir`, `tar` and `wget`) and to
+avoid hardcoding the <abbr>output</abbr> paths.
+
+The second stage trains a model using the downloaded dataset and depends on the
+rest of parameters:
+
+<toggle>
+<tab title="Python API">
+
+We can use the `dvc.api.params_show()` method in order to load all the required
+parameters inside the stage:
+
+```yaml
+  ...
+  train:
+    cmd: python train.py
+    deps:
+      - ${dataset.output_folder}
+    params:
+      - train
+```
+
+```python
+# train.py
+import dvc.api
+
+train_params = dvc.api.params_show("train")
+
+...
+```
+
+</tab>
+<tab title="Language Agnostic">
```

Comment:
Should we at least mention the option of reading in the parameters in the code using a yaml parser?